pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -158,6 +155,31 @@ func warnDeprecatedAPIs(spec *v1.VirtualMachineInstanceSpec, config *virtconfig.
 	return warnings
 }
 
+func ValidateVirtualMachineInstancePerArch(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, config *virtconfig.ClusterConfig) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+
+	switch {
+	case webhooks.IsAMD64(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceAmd64Setting(field, spec)...)
+	case webhooks.IsS390X(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceS390XSetting(field, spec)...)
+	case webhooks.IsARM64(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceArm64Setting(field, spec)...)
+	default:
+		arch := spec.Architecture
+		if arch == "" {
+			arch = config.GetConfig().ArchitectureConfiguration.DefaultArchitecture
+		}
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("unsupported architecture: %s", arch),
+			Field:   field.Child("architecture").String(),
+		})
```

Comment:
Don't we need an else here?

```suggestion
		if arch == "" {
			arch = config.GetConfig().ArchitectureConfiguration.DefaultArchitecture
		}
		else {
			causes = append(causes, metav1.StatusCause{
				Type:    metav1.CauseTypeFieldValueInvalid,
				Message: fmt.Sprintf("unsupported architecture: %s", arch),
				Field:   field.Child("architecture").String(),
			})
		}
```

Reply:
Or even better: add an `""` case.
```go
case "":
	// validation is unneeded, using default architecture
```